apps/institution-service/src/middlewares/validationMiddleware.ts

```diff
@@ -115,45 +115,81 @@ export const validateUserCanEditAggregatorIntegration = async (
   res: Response,
   next: NextFunction,
 ) => {
-  const canUserEditAggregatorIntegration =
-    await editAggregatorIntegrationValidation({
+  const validateUserCanEdit =
+    createValidateUserCanDoActionOnAggregatorIntegration({
+      adminPermission: UiUserPermissions.UPDATE_AGGREGATOR_INTEGRATION,
+      aggregatorPermission:
+        UiUserPermissions.UPDATE_AGGREGATOR_INTEGRATION_AS_AGGREGATOR,
+    });
+  await validateUserCanEdit(req, res, next);
+};
+
+export const validateUserCanDeleteAggregatorIntegration = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const validateUserCanDelete =
+    createValidateUserCanDoActionOnAggregatorIntegration({
+      adminPermission: UiUserPermissions.DELETE_AGGREGATOR_INTEGRATION,
+      aggregatorPermission:
+        UiUserPermissions.DELETE_AGGREGATOR_INTEGRATION_AS_AGGREGATOR,
+    });
+  await validateUserCanDelete(req, res, next);
+};
+
+const createValidateUserCanDoActionOnAggregatorIntegration =
+  ({
```

Comment:
The way this is written there are 2 different places that we have to keep in sync. I'd accept a validate function and import the one we already created in permissionValidation.